scripts/StateManager.js

```diff
@@ -0,0 +1,168 @@
+// Manages the current state of the extension, views can register to it
+// and it will notify certain events.
+
+'use strict';
+
+function StateManager() {
+
+	var self = this;
+
+	var options = {};
+	var callbacks = {};
+
+	var timeKeeper = new TimeKeeper();
+	var isKKTime;
+	
+	var weatherRain = ['Thunderstorm', 'Drizzle', 'Rain', 'Mist'];
+	var weatherSnow = ['Snow', 'Fog'];
+	var weather = "Clear";
+
+	this.registerCallback = function(event, callback) {
+		callbacks[event] = callbacks[event] || [];
+		callbacks[event].push(callback);
+	};
+
+	this.getOption = function(option) {
+		return options[option];
+	};
+
+	this.activate = function() {
+		isKKTime = timeKeeper.getDay() == 6 && timeKeeper.getHour() >= 20;
+		getSyncedOptions(function() {
+			notifyListeners("volume", [options.volume]);
+			if (isKK()) {
+				notifyListeners("kkStart");
+			}
+			else if(isLive()) {
+				if(options.music == 'new-leaf-live') {
+					updateWeatherCond(options.zipCode, options.countryCode, function(response) {
```

Comment:
It would be cleaner to make WeatherManager its own object with a callback for when the weather changes, like TimeKeeper is. This makes the code difficult to read.